services/crypto_get_account_records.proto

```diff
@@ -43,6 +44,25 @@ message CryptoGetAccountRecordsQuery {
      * The account ID for which the records should be retrieved
      */
     AccountID accountID = 2;
+
+    /**
+     * If set, a restriction on the first payer record to be returned. The queried node will 
+     * return this record (assuming one qualifies), and up to to the next 179 records, in 
+     * chronological order of consensus time.
+     *
+     * Without a restriction, the node will return the most recent payer records (in
+     * chronological order, up to a maximum of 180 records).
+     */
+    oneof starting_at {
+        /**
+        * The transaction id of the first record to return.
+        */
+        TransactionID transaction_id = 3;
+        /**
+        * The earliest consensus time of the first record to return.
+        */
+        Timestamp earliest_consensus_time = 4;
```

Comment:
Is there any difference in performance depending on which of these two is used? In both cases are we going to iterate through the records looking for the first that meets the criteria?

Reply:
Yes, with current FCQ implementation, we need to iterate through all records no matter what. (If random access was supported, we could do binary search for earliest consensus time, but it isn't.)

Reply:
Not sure we should add this feature until the data structure has a performant way to allow it...

Reply:
The current implementation already iterates through all records. 😬

Reply:
Yeah, but we're adding a limit now so it will only iterate through the first K records where K is the limit and N is the total number of records. But with this feature they could pick the last transaction id so it will be O(N).

Reply:
Oh, unless it has to iterate through all to filter down by account ID. I'm guessing it's not in a map by account id.

Reply:
Right, these records are in an `FCQueue` which doesn't even support random access or reverse traversal.

No matter how you restrict (or don't restrict) the query, we must iterate through the entire FCQ.